Enable concat support through inductor using pointwise kernels

nandesuka opened the pull request:
Summary: Add ability to always force pointwise kernel for concat codegen through Inductor.

Differential Revision: D66669372




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

blaine-rister reviewed (approved):
LGTM! I left one suggested change internally.

nandesuka committed:
Enable concat support through inductor using pointwise kernels (#141966)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/141966

Add ability to always force pointwise kernel for concat codegen through Inductor.

Reviewed By: eellison, blaine-rister

Differential Revision: D66669372
